File: frontend/src/app/car.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Car } from './car';

@Injectable({
  providedIn: 'root'
})
export class CarService {

private baseUrl = "http://localhost:8080/api/cars"

  constructor(private http : HttpClient) { }

  getCars(id: number): Observable<Object> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  createCar(car : Object): Observable<Object> {
    return this.http.post(`${this.baseUrl}` + `/create`, car);
  }

  updateCar(id: number, value: any): Observable<Object> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }

  deleteCar(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  getCarsList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }

  getCarsByModel(model: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/model/${model}`);
  }

  getCarsByTarga(targa: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/targa/${targa}`);
  }

  getCarsByCilindrata(cilindrata: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/cilindrata/${cilindrata}`);
  }

  deleteAll(): Observable<any> {
    return this.http.delete(`${this.baseUrl}` + `/delete`, { responseType: 'text' });
  }
}

File: backend/src/main/java/com/javasampleapproach/springrest/postgresql/service/CarService.java
package com.javasampleapproach.springrest.postgresql.service;

import java.util.List;

import com.javasampleapproach.springrest.postgresql.dto.Car;


public interface CarService {
	
	public List<Car> getAllCar();
	
	public Car createCar(Car car);
	
	public Long deleteCar(long id);
	
	public void deleteAllCar();
	
	public List<Car> findByModel(String model);
	
	public List<Car> findByTarga(String targa);
	
	public List<Car> findByCilindrata(int cilindrata);
	
	public Car updateCar(long id, Car car);
}
